ffmpeg/transcoder.c

```diff
@@ -270,6 +296,15 @@ int transcode(struct transcode_thread *h,
     av_packet_unref(&ipkt);
   }
 
+  if (ictx->transmuxing) {
+    for (i = 0; i < nb_outputs; i++) {
+      av_interleaved_write_frame(outputs[i].oc, NULL); // flush muxer
+      avformat_close_input(&ictx->ic);
+      ictx->ic = NULL;
+    }
+    return 0;
```

Comment:
Any particular reason why `transcode_cleanup:` is skipped for transmuxing?

Reply:
If we just want to prevent closing the muxer contexts, we can add a conditional there like we do for GPU - unless you're sure that we can skip all the things in that cleanup section.

Reply:
>  we can add a conditional there like we do for GPU 

Wanted to be clear about closing in transmuxing mode - without that conditions will be scattered around code.
In transmuxing mode, we don't open everything that's gets closed in the `transcode_cleanup`.